driver/result.go

```diff
@@ -0,0 +1,80 @@
+// Copyright 2020-2021 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package driver
+
+import (
+	"errors"
+	"io"
+
+	"github.com/dolthub/go-mysql-server/sql"
+)
+
+// Result is the result of a query execution.
+type Result struct {
+	ctx    *sql.Context
+	rows   sql.RowIter
+	result *sql.OkResult
+}
+
+func (r *Result) ok() error {
+	if r.result != nil {
+		return nil
+	}
+
+	defer func() {
+		r.rows.Close(r.ctx)
```

Comment:
Since `go-mysql-server` doesn't actually execute the query (update/insert/etc) until the row iterator is iterated, I decided to move the "find the result" logic into a function, and call it directly from `exec`. As part of that, I made sure that `Close()` is always called, unless `Next()` returns an error other than EOF.

In the case that no OkResult is found, I return a result that returns "not found".